lang_GENERIC/analyze/AST_to_IL.ml

```diff
@@ -52,61 +54,58 @@ let empty_env () = {
 (*****************************************************************************)
 (* Error management *)
 (*****************************************************************************)
-(*s: function [[AST_to_IL.error]] *)
-let error tok s =
-  raise (Parse_info.Ast_builder_error (s, tok))
-(*e: function [[AST_to_IL.error]] *)
-
-(*s: function [[AST_to_IL.warning]] *)
-let warning tok s =
-  pr2 (spf "%s: %s" (Parse_info.string_of_info tok) s)
-(*e: function [[AST_to_IL.warning]] *)
-
-(*s: function [[AST_to_IL.error_any]] *)
-let error_any any_generic msg =
-  let toks = Lib_AST.ii_of_any any_generic in
-  let s = AST_generic.show_any any_generic in
-  error (List.hd toks) (spf "%s: %s" msg s)
-(*e: function [[AST_to_IL.error_any]] *)
-
-(*s: function [[AST_to_IL.sgrep_construct]] *)
-let sgrep_construct any_generic =
-  error_any any_generic "Sgrep Construct"
-(*e: function [[AST_to_IL.sgrep_construct]] *)
 
-exception Todo of G.any
+exception Fixme of fixme_kind * G.any
 
 (*s: function [[AST_to_IL.todo]] *)
-let todo gany =
-  raise (Todo gany)
+let todo any_generic =
+  raise (Fixme(ToDo, any_generic))
 (*e: function [[AST_to_IL.todo]] *)
 
+(*s: function [[AST_to_IL.sgrep_construct]] *)
+let sgrep_construct any_generic =
+  raise (Fixme(Sgrep_construct, any_generic))
+(*e: function [[AST_to_IL.sgrep_construct]] *)
+
 (*s: function [[AST_to_IL.impossible]] *)
 let impossible any_generic =
-  error_any any_generic "Impossible Construct"
+  raise (Fixme(Impossible, any_generic))
 (*e: function [[AST_to_IL.impossible]] *)
 
+let locate opt_tok s =
+  match map_opt Parse_info.string_of_info opt_tok with
+  | Some loc -> spf "%s: %s" loc s
+  | None     -> s
+  | exception Parse_info.NoTokenLocation _ -> s
```

Comment:
hmm, I hope our OCaml parser in semgrep will parse this.

Reply:
Should I rewrite it?

Reply:
OK I removed it just in case.